src/internal/passwordHash.ts

```diff
@@ -0,0 +1,12 @@
+import * as Scrypt from "@noble/hashes/scrypt"
+import * as Data from "effect/Data"
+import * as Effect from "effect/Effect"
+
+class ScryptOptionsError extends Data.Error<{ message: string }> {}
+
+export const scrypt = (input: string | Uint8Array, salt: string | Uint8Array, options: Scrypt.ScryptOpts) =>
+  Effect.try({
+    try: () => Scrypt.scrypt(input, salt, options),
+    catch: (error) =>
```

Comment:
```ts
catch: (cause) => new ScryptOptionsError({ cause })
```

Reply:
Doesn't work this way?

<img width="921" alt="Screenshot 2024-01-08 at 22 18 27" src="https://github.com/fubhy/effect-crypto/assets/223045/a7cd02af-b1b3-498c-b68b-9631bfc6fef7">